CompilerSource/parser/collect_variables.cpp

```diff
@@ -236,7 +236,8 @@ class DeclGatheringVisitor : public AST::Visitor {
       dectrip dtrip(type, prefix, suffix);
       if (is_global) parsed_scope->globals[name] = dtrip;
       if (is_local) parsed_scope->locals[name] = dtrip;
-      parsed_scope->declarations[name] = dtrip;
+      cs->add_dot_accessed_local(name);
```

Comment:
It is useful for scripts, as it allows us to have `var &varaccess_foo(int x);` for all variables, enabling access to variables from within scripts using `enigma::varaccess_foo(int(self))`.